modules/skybrowser/src/targetbrowserpair.cpp

```diff
@@ -0,0 +1,384 @@
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/skybrowser/include/targetbrowserpair.h>
+
+#include <modules/skybrowser/include/screenspaceskybrowser.h>
+#include <modules/skybrowser/include/screenspaceskytarget.h>
+#include <modules/skybrowser/include/utility.h>
+#include <modules/skybrowser/include/wwtdatahandler.h>
+#include <openspace/engine/globals.h>
+#include <openspace/rendering/screenspacerenderable.h>
+#include <openspace/scripting/scriptengine.h>
+#include <ghoul/misc/assert.h>
+#include <glm/gtc/constants.hpp>
+#include <functional>
+
+namespace openspace {
+
+    TargetBrowserPair::TargetBrowserPair(ScreenSpaceSkyBrowser* browser, 
+                                         ScreenSpaceSkyTarget* target)
+       : _target(target), _browser(browser)
+    {
+        ghoul_assert(browser != nullptr, "Sky browser is null pointer!");
+        ghoul_assert(target != nullptr, "Sky target is null pointer!");
+    }
+
+    TargetBrowserPair& TargetBrowserPair::operator=(TargetBrowserPair other) {
+        std::swap(_target, other._target);
+        std::swap(_browser, other._browser);
+        return *this;
+        
+    }
+    
+    void TargetBrowserPair::lock() {
+        _target->setLock(true);
+    }
+
+    void TargetBrowserPair::unlock() {
+        _target->setLock(false);
+    }
+
+    void TargetBrowserPair::setImageOrder(int i, int order) {
+        _browser->setImageOrder(i, order);
+    }
+
+    void TargetBrowserPair::removeHighlight(glm::ivec3 color) {
+        _target->removeHighlight(color);
+        _browser->removeHighlight(color);
+    }
+
+    void TargetBrowserPair::highlight(glm::ivec3 color) {
+        _browser->highlight(color);
+        _target->highlight(color);
+    }
+
+    bool TargetBrowserPair::isTargetFadeFinished(float goalState) const {
+        // Is fading finished?
+        float targetDiff = abs(_target->opacity() - goalState);
+        
+        return targetDiff < FadeThreshold;
+    }
+
+    bool TargetBrowserPair::isBrowserFadeFinished(float goalState) const {
+        float browserDiff = abs(_browser->opacity() - goalState);
+        return browserDiff < FadeThreshold;
+    }
+
+    bool TargetBrowserPair::checkMouseIntersection(glm::vec2 mousePosition) {
+        bool onBrowser = _browser->intersection(mousePosition);
```

Comment:
The `intersection` function needs a more describing, "verb-y" name. maybe `testIntersection`?